rfcs/0001-reason-message-diff.md

```diff
@@ -2,52 +2,291 @@
 
 **For Reason "CONFIG"**
-- Ex. Reason: CONFIG - The [image-name] image config was updated. See diff below.
-[output of diffing library on the two image config files]
 
-**For Reason "BUILDPACK"**
-- Ex. Reason: BUILDER - The buildpacks used by your app have changed. See diff below. 
-[output of diffing library on the two Bill of Materials' for the builds]
+- Print a diff beneath the reason section
+- Print the three lines preceding where the diff begins in order to show context for the change. 
+
+------------------------------
+
+$kp build status petclinic
+Image:      index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+Status:     SUCCESS
+Reasons:    CONFIG
+             Build:
+                env:
+                - name: BP_JAVA_VERSION
+            - value: 11.*
+            - resources: {}
+            - CacheSize: 1M
+            + value: 14.*
+            + resources:
+            +   limits:
+            +       cpu: 1M
+            + CacheSize: 12M
+
+Builder:    our-special-url.jfrog.io/instance/cluster-builder:default@sha4398ths9rdpfp9sghbne985-s9ed9hgbd9rz4sh98dxbfs
+Run Image:  index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser89
+
+Source:   Local Source 
+
+BUILDPACK ID                            BUILDPACK VERSION
+paketo-buildpacks/bellsoft-liberica     2.13.0
+paketo-buildpacks/maven                 2.3.0
+paketo-buildpacks/executable-jar        2.1.1
+paketo-buildpacks/apache-tomcat         1.5.0
+paketo-buildpacks/dist-zip              1.4.0
+paketo-buildpacks/spring-boot           2.5.0
+
+------------------------------
 
 **For Reason "STACK"**
-- Ex. Reason: STACK - Stack-name changed from dockerhub/sha1 to dockerhub/sha2
 
-**For Reason "TRIGGER"**
-- Ex. Reason: TRIGGER - A new build was manually triggered by [user] at [time]
+- Print a diff beneath the reason section
+- Print the stack-name after the reason in order to give context 
 
-### Prior Art:
-We have been thinking about different ways to surface the [Bill of Materials](https://github.com/buildpacks/pack/issues/378) for a build and how to show the delta between builds for a few months now. This is an evolution of that work. Although the `pack` cli renders the contents of a single BoM alone with the `inspect-image` command, I believe we don't need to provide a BoM alone, because the purpose of kpack is to iteratively build images, and therefore build BoMs should be placed in context as much as possible.
+------------------------------
 
-### Complexity/Risks:
-How does the output render when there are two Reasons for a new build?
+$kp build status petclinic
+Image:      index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+Status:     SUCCESS
+Reason:     STACK
+            - RunImage: index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser892
```

Comment:
Out of scope here, but makes me think about how valuable it would be to have a stack homepage link so the user could review what changed about the stacks rather than just see the diff in the digest.

Reply:
yea totally! 
we should do a brainstorm with the core deps team about it

Reply:
+1. Would be super cool if users could see the dpkg output difference